aspnetcore/security/authentication/azure-ad-b2c-webapi.md

```diff
@@ -0,0 +1,208 @@
+| **Include web app / web API** | Yes                 |                                                                                        |
+| **Allow implicit flow**       | Yes                 |                                                                                        |
+| **Reply URL**                 | `https://localhost` | Reply URLs are endpoints where Azure AD B2C returns any tokens that your app requests. |
+| **App ID URI**                | *api*               | The URI doesn't need to resolve to a physical address. It only needs to be unique.     |
+| **Include native client**     | No                  |                                                                                        |
+
+After the API is registered, the list of apps and APIs in the tenant is displayed. Select the API that was just registered. Select the **Copy** icon to the right of the **Application ID** field to copy the Application ID to the clipboard.  Select **Published scopes** and verify the default *user_impersonation* scope is present.
+
+## Create an ASP.NET Core app in Visual Studio 2017
+
+The Visual Studio Web Application template can be configured to use the Azure AD B2C tenant for authentication.
+
+In Visual Studio:
+
+1. Create a new ASP.NET Core Web Application. 
+2. Select **Web API** from the list of templates.
+3. Select the **Change Authentication** button.
+    
+    ![Change Authentication Button](./azure-ad-b2c-webapi/change-auth-button.png)
+
+4. In the **Change Authentication** dialog, select **Individual User Accounts**, and then select **Connect to an existing user store in the cloud** in the dropdown. 
+    
+    ![Change Authentication Dialog](./azure-ad-b2c-webapi/change-auth-dialog.png)
+
+5. Complete the form with the following values:
+    
+    | Setting                       | Value                                                 |
+    |-------------------------------|-------------------------------------------------------|
+    | **Domain Name**               | *&lt;the domain name of your B2C tenant&gt;*          |
+    | **Application ID**            | *&lt;paste the Application ID from the clipboard&gt;* |
+    | **Sign-up or sign-in policy** | `B2C_1_SiUpIn`                                        |
+    
+    Select **OK** to close the **Change Authentication** dialog. Select **OK** to create the web app.
+
+Visual Studio creates the web API with a controller named *ValuesController.cs* that returns hard-coded values for GET requests.  The class is decorated with the [Authorize attribute](xref:security/authorization/simple), so all requests require authentication.
+
+## Run the web API
+
+In Visual Studio, press *F5* to run the API. Visual Studio launches a browser pointed at the APIs root URL. Note the URL in the address bar, and leave the API running in the background.
```

Comment:
I may be best not to recommend a method for running the API. Readers may wonder why they can't click the run button instead. Maybe just say "In Visual Studio, run the API."